0xProject/0x-monorepo
# [WIP] [sol-resolver] Fix *nix-like hardcoded paths in NPM Resolver

## Description

The NPM resolver currently assumes a hardcoded root directory
of '/' and a path separator of '/'.

Currently it uses a while loop to recursively move up directories
from the current project working directory to the root directory,
searching for dependencies in possible node_modules locations.

The while loop breaks if 'trial directory === the hardcoded root'
i.e. trial as a result of moving up directories  === '/'

On windows machines this statement will never be true, as the root
path of the file system is 'X:/' (where X is any letter [A-Z]), thus this 
loop is infinitely recursive.

Additionally, it hardcodes a path separator of '/' which on Windows
is instead '\'.

This fix utilises Node's path library to gather the current system root 
directory and separator.

It uses 'path.parse(...).root' for the system root and 'path.sep'.

## Testing instructions

- Use `sol-compiler --contracts-dir contracts` on Windows and *nix
    - _Previous_: Observe recursive hang on Windows
    - _Now_: Contracts compile correctly on Windows

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist:
